.github/workflows/tutorials.yml

```diff
@@ -0,0 +1,81 @@
+name: Jupyter Notebooks
+
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the master branch
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  tutorials:
+    name: Test Jupyter notebooks on ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+
+    env:
+      DEVITO_ARCH: "${{ matrix.compiler }}"
+      DEVITO_OPENMP: 1
+      DEVITO_BACKEND: "core"
+      PYTHON_VERSION: "3.7"
+
+    strategy:
+      # Prevent all build to stop if a single one fails
+      fail-fast: false
+      matrix:
+        name: [
+          ubuntu-16.04-gcc-py37,
+          macOS-10.14-gcc-py37
+          ]
+
+        include:
+          - name: ubuntu-16.04-gcc-py37
+            os: ubuntu-16.04
+            compiler: gcc-7
+
+          - name: macOS-10.14-gcc-py37
+            os: macOS-10.14
+            compiler: gcc-9
+
+    steps:
+    - name: Checkout devito
+      uses: actions/checkout@v1
+
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+
+    - name: Install compilers
+      run: |
+        if [ "${{ matrix.compiler }}" = "gcc-7" ]; then
+           sudo apt-get install -y mpich libmpich-dev
+        else
+           brew install gcc
+           brew install mpich
+        fi
+
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip
+        pip install -e .[extras]
+
+    - name: CFD notebooks
```

Comment:
I need to be able to call these tests from within the Docker testing workflow. Splitting them like this means this entire thing needs to repeated for the other workflow. Obviously not a good idea.

Reply:
Separating in step and being able to see and maintain the CI is the whole point of actions, I'm not stashing stuff in some script "for docker" or what's the point of all this, just put everything in a single bash script. This is clear, readable and maintainable.

Reply:
Copying and pasting the same workflow to multiple places is a clear violation of the DRY principle. What principle is being violated by putting things in a shell script?

Reply:
From slack discussions, the concern is that when there is a failure within a bash script the source of the error is not clear from the Github-action logs. I feel there is still an issue with *not* using bash scripts as we might lose out on some code reuse. Perhaps there needs to be another way of composing actions to avoid this issue, or perhaps there is already a way we have overlooked as we are all Github novices. 

I am going to mark this thread as resolved for now but I expect the to-bash-or-not-to-bash issue is going to arise again and we should continue to reconsider the question as Github-actions matures.